artemis/fallback_api_cache.py

```diff
@@ -0,0 +1,93 @@
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Any, Callable, Dict, List
+
+from requests import Response
+from requests_cache import CachedSession
+from requests_cache.backends.filesystem import FileCache
+
+from artemis import utils
+from artemis.config import Config
+
+
+class InvalidResponseException(Exception):
+    pass
+
+
+@dataclass
+class CachedURL:
+    url: str
+    validator: Callable[[Dict[Any, Any]], bool]
+
+    def get(self) -> Response:
+        return FallbackAPICache.get(self.url)
+
+
+class FallbackAPICache:
+    LOGGER = utils.build_logger(__name__)
+
+    CACHE = CachedSession(
+        backend=FileCache("/cache"),
+        cache_control=False,
+        expire_after=timedelta(seconds=3600),
+        allowable_codes=[200],
+        stale_if_error=True,
+    )
+
+    URL_JOOMLA_LATEST_RELEASE = CachedURL(
```

Comment:
I am not fan of using cls fields and then iterating over them via `dir(cls)`. Change these cls fields into one cls field typed `Dict[CachedURL]` and iterate over it